flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeSerdeTest.java

```diff
@@ -18,58 +18,485 @@
+
+        // Module provided permanent function
+        serdeContext
+                .getFlinkContext()
+                .getModuleManager()
+                .loadModule("myModule", FunctionProvidingModule.INSTANCE);
+        callable.call();
+    }
+
+    @Test
+    public void testTemporarySystemFunction() throws Throwable {
+        final SerdeContext serdeContext = contradictingSerdeContext();
+
+        final ThrowingCallable callable =
+                () ->
+                        testJsonRoundTrip(
+                                serdeContext,
+                                createFunctionCall(
+                                        serdeContext,
+                                        ContextResolvedFunction.temporary(
+                                                FUNCTION_SYS_ID, NON_SER_UDF_IMPL)),
+                                RexNode.class);
+
+        // Missing function
+        assertThatThrownBy(callable)
+                .satisfies(
+                        anyCauseMatches(
+                                TableException.class,
+                                "Could not lookup system function '" + FUNCTION_NAME + "'."));
+
+        // Registered temporary function
+        registerTemporaryFunction(serdeContext);
+        callable.call();
+    }
+
+    @Test
+    public void testTemporaryCatalogFunction() throws Throwable {
+        final SerdeContext serdeContext = contradictingSerdeContext();
+
+        final ThrowingCallable callable =
+                () ->
+                        testJsonRoundTrip(
+                                serdeContext,
+                                createFunctionCall(
+                                        serdeContext,
+                                        ContextResolvedFunction.temporary(
+                                                FUNCTION_CAT_ID, NON_SER_FUNCTION_DEF_IMPL)),
+                                RexNode.class);
+
+        // Missing function
+        assertThatThrownBy(callable)
+                .satisfies(
+                        anyCauseMatches(
+                                TableException.class,
+                                "The persisted plan does not include all required "
+                                        + "catalog metadata for function '"
+                                        + FUNCTION_CAT_ID.asSummaryString()
+                                        + "'."));
+
+        // Registered temporary function
+        registerTemporaryFunction(serdeContext);
+        callable.call();
+    }
+
+    @Nested
+    @DisplayName("Test CatalogPlanCompilation == IDENTIFIER")
+    class TestCompileIdentifier {
+
+        private final CatalogPlanCompilation compilation = CatalogPlanCompilation.IDENTIFIER;
+
+        @Nested
+        @DisplayName("and CatalogPlanRestore == IDENTIFIER")
+        class TestRestoreIdentifier {
+
+            private final CatalogPlanRestore restore = CatalogPlanRestore.IDENTIFIER;
```

Comment:
maybe avoid recreating the ctx everytime but just create one here?

Reply:
the tests modify the ctx via `dropPermanentFunction` or `registerTemporaryFunction`, it is safer to create a new one per test